### ISSUE TITLE
[BUG] In-flight Analysis of key rotation policy

### ISSUE THREAD

**Issue by Feylin** [Azure/PSRule.Rules.Azure#3261]
### Existing rule

Azure.KeyVault.AutoRotationPolicy

### Description of the issue

It seems there might be an issue when retrieving rotation policies of keys in a key vault.
The Azure.KeyVault.AutoRotationPolicy passes when performing pre-flight analysis of the resources described below, but fails when performing in-flight analysis of the deployed resources.

```yaml
include:
  # Use rules from the following modules
  module:
    - PSRule.Rules.Azure
  path:
    # This is the default search path
    - .ps-rule

configuration:
  # Set the Bicep parameter expansion configuration option to enable expansion of Azure Bicep parameter files (.bicepparam).
  AZURE_PARAMETER_FILE_EXPANSION: true
  # Enable expansion for Bicep source files.
  AZURE_BICEP_FILE_EXPANSION: true
  # Validate that the Bicep CLI is used
  AZURE_BICEP_CHECK_TOOL: true
  AZURE_BICEP_FILE_EXPANSION_TIMEOUT: 30
  AZURE_BICEP_MINIMUM_VERSION: '0.33.93'

# Require a minimum version of modules that include referenced baseline.
requires:
  PSRule: '>=2.9.0'
  PSRule.Rules.Azure: '>=1.41.3'

binding:
  targetType:
    - 'resourceType'
    - 'type'
```

### Reproduction

I have defined the following resource.

```bicep
resource keyVault 'Microsoft.KeyVault/vaults@2023-07-01' = {
  name: 'keyvaulttest'
  location: 'northeurope'
  properties: {
    enabledForDeployment: true
    enabledForTemplateDeployment: true
    enabledForDiskEncryption: true
    tenantId: subscription().tenantId
    accessPolicies: [
      {
        objectId: '00000000-0000-0000-0000-000000000000'
        permissions: {
          keys: [
            'all'
          ]
        }
        tenantId: subscription().tenantId
      }
    ]
    sku: {
      name: 'standard'
      family: 'A'
    }
  }

  resource key 'keys' = {
    name: 'secret'
    properties: {
      rotationPolicy: {
        attributes: {
          expiryTime: 'P90D'
        }
        lifetimeActions: [
          {
            action: {
              type: 'rotate'
            }
            trigger: {
              timeAfterCreate: 'P10D'
            }
          }
        ]
      }
      attributes: null
      keySize: 2048
      kty: 'RSA'
      curveName: 'P-256'
      keyOps: [
        'encrypt'
        'decrypt'
      ]
    }
  }
}
```

When analyzing it using the following command it correctly states that it passes the rotation policy rule

```powershell
(invoke-psrule -InputPath .\main.bicep) | Where-Object RuleName -EQ Azure.KeyVault.AutoRotationPolicy
```

However, when I want to analyze the deployed resources using the following commands

```powershell
export-azruledata -ResourceGroupName <resourcegroupname>
```
```powershell
(invoke-psrule -InputPath <jsonfile>) | Where-Object RuleName -EQ Azure.KeyVault.AutoRotationPolicy
```

It fails the same aforementioned rule.

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.41.3

### Additional context

I thought at first it might have something to do with the user context when running the commands, but it is in a sandbox environment where I have all the permissions in the world, and I can easily get the rotation policies using the azure cli and the azure cmdlets.

Hopefully it is not a limitation of some sort.

**Comment by Feylin** [comment]
Thank you for letting me know and for looking into it. I'll be sure to keep an eye on this issue in the coming future.

**Comment by BernieWhite** [comment]
@Feylin Currently there isn't a workable solution to this for in-flight analysis which currently only uses the management plane as mentioned before. So, we're going to disable export of `keys` so that the rule will no longer trigger in the in-flight scenario.

In a future iteration, we'll consider adding a separate export option to export this information again from the data plane with docs for customers that want to do it and are geared to support the permission and network access requirements for private endpoints or VNET access as they are configured.
